api/v1beta1/types.go

```diff
@@ -589,9 +592,20 @@ type SubnetSpec struct {
 	// +optional
 	NatGateway NatGateway `json:"natGateway,omitempty"`
 
+	// ServiceEndpoints is a slice of Virtual Network service endpoints to enable for the subnets.
+	// +optional
+	ServiceEndpoints `json:"serviceEndpoints,omitempty"`
```

Comment:
Is there an argument that this is more clear?:

```
// ServiceEndpoints is a slice of Virtual Network service endpoints to enable for the subnets.
// +optional
ServiceEndpoints ServiceEndpoints `json:"serviceEndpoints,omitempty"`
```

@nojnhuh @CecileRobertMichon  any thoughts here? A cursory look suggests that other slice types in this file (`Subnets`, `VnetPeerings`) are not embedded like the above, but have an explicit property name when used in other type definitions.

AFAIK in golang the two options are exactly equivalent, but for my combining the golang struct embedding pattern w/ slice types is a bit harder to grok.

Reply:
I tend to agree that if we don't need the hoisting-like attributes of embedded properties, then giving an explicit name might be a bit more clear. It might not be as big of a deal in this particular since Go doesn't let you [index into a type that embeds a slice](https://go.dev/play/p/b7HYbUtuTkm) which would likely be surprising to read, but I can see value in naming the field for consistency's sake.

Reply:
This is an oversight on my part. I meant to do `ServiceEndpoints ServiceEndpoints `json:"serviceEndpoints,omitempty"`` like I did for the managed control plane.